components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/SystemNotificationBridge.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.webnotifications
+
+import android.app.Notification
+import android.content.Context
+import android.graphics.Bitmap
+import android.net.Uri
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.browser.icons.Icon.Source
+import mozilla.components.browser.icons.IconRequest
+import mozilla.components.browser.icons.IconRequest.Size
+import mozilla.components.concept.engine.webnotifications.WebNotification
+
+internal class SystemNotificationBridge(
+    private val icons: BrowserIcons
+) {
+
+    /**
+     * Create a system [Notification] from this [WebNotification].
+     */
+    @Suppress("LongMethod")
+    suspend fun convertToAndroidNotification(
+        notification: WebNotification,
+        context: Context,
+        channelId: String
+    ): Notification {
+        val builder = if (SDK_INT >= Build.VERSION_CODES.O) {
+            Notification.Builder(context, channelId)
+        } else {
+            @Suppress("Deprecation")
+            Notification.Builder(context)
+        }
+
+        with(notification) {
+            loadIcon(iconUrl, origin, Size.DEFAULT)?.let { icon ->
+                builder.setLargeIcon(it)
+            }
+
+            builder.setContentTitle(title).setContentText(body)
+
+            @Suppress("Deprecation")
+            builder.setVibrate(vibrate)
+
+            timestamp?.let {
+                builder.setShowWhen(true).setWhen(it)
+            }
+
+            if (silent) {
+                @Suppress("Deprecation")
+                builder.setDefaults(0)
+            }
```

Comment:
I think properties like silent and vibrate will be ignored if a notification channel is used. Probably nothing we can do here though. Still makes sense to set the behavior for older versions.

Reply:
Yeah, we can at least support older devices here.